src/MobManager.cpp

```diff
@@ -751,16 +749,11 @@ void MobManager::retreatStep(Mob *mob, time_t currTime) {
 
 void MobManager::step(CNServer *serv, time_t currTime) {
     for (auto& pair : Mobs) {
-
-        // skip chunks without players
-        if (pair.second->playersInView == 0) //(!ChunkManager::inPopulatedChunks(pair.second->viewableChunks))
-            continue;
-
         if (pair.second->playersInView < 0)
             std::cout << "[WARN] Weird playerview value " << pair.second->playersInView << std::endl;
 
-        // skip mob movement and combat if disabled
-        if (!simulateMobs && pair.second->state != MobState::DEAD
+        // skip mob movement and combat if disabled or not in view
+        if ((!simulateMobs || pair.second->playersInView == 0) && pair.second->state != MobState::DEAD
```

Comment:
I'm actually not a fan of moving the view check down here, because that means the mob could be simulated even if there's no players in view, which shouldn't _ever_ happen.

Reply:
I'm of the opinion a mob should simulate when dead, otherwise the mob will not respawn when a player is not closeby.

Reply:
That's a good point. We'll leave it and see if performance holds.